src/main/java/fi/otavanopisto/kuntaapi/server/discover/IdUpdater.java

```diff
@@ -104,38 +105,42 @@ public int getTimerInterval() {
       }
     }
     
-    return 1000 * 60;
-  }
-
-  private void startTimer(int duration) {
-    TimerConfig timerConfig = new TimerConfig();
-    timerConfig.setPersistent(false);
-    getTimerService().createSingleActionTimer(duration, timerConfig);
+    return 1000l * 10;
   }
   
-  @Timeout
-  public void onTimeout() {
-    if (!isStopped()) {
+  private void startTimer(long warmup, long delay) {
+    managedScheduledExecutorService.scheduleWithFixedDelay(() -> {
+      UserTransaction userTransaction = ejbContext.getUserTransaction();
       try {
-        if (isEligibleToRun()) {
+        //userTransaction = lookup();
+        userTransaction.begin();
+        if (!isStopped() && isEligibleToRun()) {
+          logger.log(Level.INFO, String.format("Running timer %s", getName()));
           timeout();
         }
-      } catch (Exception e) {
-        logger.log(Level.SEVERE, "Timer throw an exception", e);
-      } finally {
+        userTransaction.commit();
+      } catch (Exception ex) {
+        logger.log(Level.SEVERE, String.format("Timer with name %s throw an exception", getName()), ex);
         try {
-          Iterator<Timer> timers = getTimerService().getTimers().iterator();
-          while(timers.hasNext()) {
-            Timer timer = timers.next();
-            timer.cancel();
+          if(userTransaction != null) {
+              userTransaction.rollback();
           }
-        } catch (Exception e) {
-          logger.log(Level.SEVERE, "Exception while canceling timer", e);
-        } finally {
-          startTimer(getTimerInterval());
+        } catch (SystemException e1) {
+          logger.log(Level.SEVERE, "Failed to rollback transaction", e1);
         }
       }
+    }, warmup, delay, TimeUnit.MILLISECONDS);
+  }
+  
+  private UserTransaction lookup() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "lookup" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)